internal/openstack/client.go

```diff
@@ -223,8 +223,20 @@ func getProvider(
 	authInfo.AuthURL = authURL
 	authInfo.Username = strings.TrimSpace(cfg.Section("Global").Key("username").String())
 	authInfo.Password = strings.TrimSpace(cfg.Section("Global").Key("password").String())
+
 	authInfo.DomainName = strings.TrimSpace(cfg.Section("Global").Key("domain-name").String())
-	authInfo.ProjectName = strings.TrimSpace(cfg.Section("Global").Key("tenant-name").String())
+	authInfo.DomainID = strings.TrimSpace(cfg.Section("Global").Key("domain-id").String())
+
+	legacyProjectName := strings.TrimSpace(cfg.Section("Global").Key("tenant-name").String())
+	projectName := strings.TrimSpace(cfg.Section("Global").Key("project-name").String())
+	authInfo.ProjectID = strings.TrimSpace(cfg.Section("Global").Key("project-id").String())
+
+	// TODO: remove with major version bump
```

Comment:
I think we should not remove it that fast. It depends on how you get the secrets and how they look.
```suggestion
```

Reply:
tenant-name has been legacy for years now. I don't know if it is worth keeping

Reply:
had the discussion in gardener before already... https://github.com/gardener/gardener-extension-provider-openstack/pull/441

we should wait at least until https://github.com/kubernetes/cloud-provider-openstack/blob/master/docs/openstack-cloud-controller-manager/using-openstack-cloud-controller-manager.md#global they managed to switch to project name instead of tenant name

Reply:
@breuerfelix That is a different matter entirely. We have been setting `tenant-name` to (`authInfo.ProjectName`) since basically the start of this Project. The only difference is that it was named in the secret with the legacy name.

We will not break anything especially if we still support the old-way.

Reply:
ye of course, i just mean that other projects are not removing it, even worse, they are not even yet supporting the "new" way with project name. and unless the official kubernetes openstack provider hasnt changed to project name, we should still support tenant name, even though i would like to remove it aswell :D

Reply:
> ye of course, i just mean that other projects are not removing it, even worse, they are not even yet supporting the "new" way with project name. and unless the official kubernetes openstack provider hasnt changed to project name, we should still support tenant name, even though i would like to remove it aswell :D

be the change you wanna see. To me it makes no sense and there is very little correlation for keeping the old-name.